cmd/servicelog/link_validator.go

```diff
@@ -0,0 +1,69 @@
+package servicelog
+
+import (
+	"fmt"
+	"net/http"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// LinkValidator handles validation of URLs in service log messages
+type LinkValidator struct {
+	timeout       time.Duration
+	skipLinkCheck bool
+	httpClient    *http.Client
+}
+
+// NewLinkValidator creates a new LinkValidator with default settings
+func NewLinkValidator(skipLinkCheck bool) *LinkValidator {
+	return &LinkValidator{
+		timeout:       time.Second * 5,
+		skipLinkCheck: skipLinkCheck,
+		httpClient:    &http.Client{Timeout: time.Second * 5},
+	}
+}
+
+// Extract URLs from service log
+func (lv *LinkValidator) extractURLs(text string) []string {
+	urlRegex := regexp.MustCompile(`https?://[^\s]+`)
+	matches := urlRegex.FindAllString(text, -1)
+
+	var cleanURLs []string
+	for _, match := range matches {
+		// Remove common trailing punctuation
+		cleanURL := strings.TrimRight(match, ".,;:!?)]}")
+		cleanURLs = append(cleanURLs, cleanURL)
+	}
+
+	return cleanURLs
+}
+
+// Check if URL is active
+func (lv *LinkValidator) checkURL(url string) error {
+	resp, err := lv.httpClient.Head(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+	if resp.StatusCode >= 400 {
+		return fmt.Errorf("HTTP %d", resp.StatusCode)
+	}
```

Comment:
If we get a 403 unauthorized, do we want to count it as a dead link? Also, if we get a 500, we probably don't want to count it, and maybe retry.